AddressBook.java

```diff
@@ -0,0 +1,1317 @@
+    	all_users.put(personProperty.EMAIL, personEmail);
+    	 
+        ALL_PERSONS.add(all_users);
+
+        savePersonsToFile(getAllPersonsInAddressBook(), storageFilePath);
+    }
+
+    /**
+     * Deletes a person from the address book, target is identified by it's absolute index in the full list.
+     * Saves changes to storage file.
+     *
+     * @param index absolute index of person to delete (index within {@link #ALL_PERSONS})
+     */
+    /* Im commenting this redundant bit of code out.
+    private static void deletePersonFromAddressBook(int index) {
+        ALL_PERSONS.remove(index);
+        savePersonsToFile(getAllPersonsInAddressBook(), storageFilePath);
+    }
+    */
+
+    /**
+     * Deletes the specified person from the addressbook if it is inside. Saves any changes to storage file.
+     *
+     * @param exactPerson the actual person inside the address book (exactPerson == the person to delete in the full list)
+     * @return true if the given person was found and deleted in the model
+     */
+    private static String deletePersonFromAddressBook(String exactPerson) {
+    	final boolean changed;
+    	HashMap<personProperty, String> store= new HashMap<personProperty, String>();
+    	for(HashMap<personProperty, String> personElement:ALL_PERSONS){
+    		if(personElement.get(personProperty.NAME).equals(exactPerson)){
+    			 store=personElement;
+    		}
+    	}
+    	changed = ALL_PERSONS.remove(store);
+        if (changed) {
+            savePersonsToFile(getAllPersonsInAddressBook(), storageFilePath);
+            return getMessageForSuccessfulDelete(store);
+        }
+        
+        return MESSAGE_PERSON_NOT_IN_ADDRESSBOOK;
+    }
+
+    /**
+     * @return unmodifiable list view of all persons in the address book
+     */
+    private static ArrayList<HashMap<personProperty, String>> getAllPersonsInAddressBook() {
+        return ALL_PERSONS;
+    }
+
+    /**
+     * Clears all persons in the address book and saves changes to file.
+     */
+    private static void clearAddressBook() {
+        ALL_PERSONS.clear();
+        savePersonsToFile(getAllPersonsInAddressBook(), storageFilePath);
+    }
+
+    /**
+     * Resets the internal model with the given data. Does not save to file.
+     *
+     * @param persons list of persons to initialise the model with
+     */
+    private static void initialiseAddressBookModel(ArrayList<HashMap<personProperty, String>> persons) {
+        ALL_PERSONS.clear();
+        ALL_PERSONS.addAll(persons);
+    }
+
+
+    /*
+     * ===========================================
+     *             PERSON METHODS
+     * ===========================================
+     */
+
+    /**
+     * @param person whose name you want
+     * @return person's name
+     */
+    private static String getNameFromPerson(HashMap<personProperty, String> person) {
+         return person.get(personProperty.NAME);
+    	/*return person[PERSON_DATA_INDEX_NAME];*/
```

Comment:
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.